### PR TITLE
Setup: Add traffic generator pod

### PR DESCRIPTION
**Pull request by RamLavi**
This PR introduces the traffic generator pod instance, which is created on the checkup's Setup function

### CODE REVIEW

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1674567780

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [30.542 seconds]
------------------------------

Ran 1 of 1 Specs in 30.548 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (30.55s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	30.580s
```

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/65bdb637fabd85547dd2aebbce3f0992fc34f81d..b44bbfe17f866272a82437e850c7b171931f6ec1): fix lint issue

**Comment by RamLavi** [comment]
Passed on CNV 4.12 machine
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1674739902

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [35.819 seconds]
------------------------------

Ran 1 of 1 Specs in 35.824 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (35.82s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	35.855s
```

File: tests/checkup_test.go [hunk 1]
```diff
@@ -223,6 +223,16 @@ func newKubeVirtDPDKCheckerRole() *rbacv1.Role {
 				Resources: []string{"virtualmachineinstances"},
 				Verbs:     []string{"create", "get", "delete"},
 			},
+			{
```
Comment (orelmisan) [inline tests/checkup_test.go:226]: In the commit message it says that the RoleBinding is also changed, while in the code it does not.

File: pkg/internal/checkup/pod/spec.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+package pod
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/spec.go:1]: Please add the license comment.

File: pkg/internal/checkup/pod/spec.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	}
+}
+
+func WithPodContainer(container *corev1.Container) PodOption {
+	return func(pod *corev1.Pod) {
+		pod.Spec.Containers = append(pod.Spec.Containers, *container)
+	}
+}
+
+func NewPod(name string, opts ...PodOption) *corev1.Pod {
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/spec.go:120]: Could you please move this function to the top of the file?

File: pkg/internal/checkup/pod/spec.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.PodSpec{},
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/spec.go:125]: Please check if this initialization is necessary.
I think its redundant because Spec is not a pointer.

File: pkg/internal/checkup/pod/container.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
+package pod
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/container.go:1]: Please add the license comment.

File: pkg/internal/checkup/pod/container.go [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	}
+}
+
+func NewSecurityContext(user int64, allowPrivilegeEscalatio bool, addCapabilities []corev1.Capability) *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		AllowPrivilegeEscalation: pointer(allowPrivilegeEscalatio),
+		Capabilities: &corev1.Capabilities{
+			Add: addCapabilities,
+		},
+		RunAsUser: pointer(user),
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/container.go:97]: Why is this needed?
The user could be defined at the Dockerfile.

File: pkg/internal/checkup/pod/multus.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
+package pod
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/multus.go:1]: Please add the licence comment.

File: pkg/internal/checkup/pod/multus.go [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+func CreateNetworksRequest(networkRequestsMap map[string]*networkv1.NetworkAttachmentDefinition) (string, error) {
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/multus.go:11]: If the `networkRequestsMap` maps MAC address to NetworkAttachmentDefinnition,
Please consider using the net.HWAddr for it, or to rename it to MACToNADMap.

File: pkg/internal/checkup/pod/pod_test.go [hunk 9]
```diff
@@ -0,0 +1,193 @@
+package pod_test
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/pod_test.go:1]: Please add license comment.

File: pkg/internal/checkup/pod/pod_test.go [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/checkup/pod"
+)
+
+const testPodName = "my-pod"
+const testContainerName = "my-container"
+
+func TestCreateNetworkRequestAnnotationShouldSucceed(t *testing.T) {
+	type successTestCase struct {
+		description              string
+		inputNetworkRequestsMap  map[string]*networkv1.NetworkAttachmentDefinition
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/pod_test.go:20]: Please consider to rename this map, in case you change the name of the production one.

File: pkg/internal/checkup/pod/pod_test.go [hunk 11]
```diff
@@ -0,0 +1,193 @@
+package pod_test
+
+import (
+	"testing"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/stretchr/testify/assert"
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/pod_test.go:7]: Please move it below the std library.

File: pkg/internal/checkup/pod/pod_test.go [hunk 12]
```diff
@@ -0,0 +1,193 @@
+package pod_test
+
+import (
+	"testing"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/pod_test.go:6]: Please move it bellow the K8s imports.

File: pkg/internal/checkup/pod/pod_test.go [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+		description              string
+		inputNetworkRequestsMap  map[string]*networkv1.NetworkAttachmentDefinition
+		expectedNetworksRequests []string
+	}
+
+	testCases := []successTestCase{
+		{
+			description:              "when the input map is empty",
+			inputNetworkRequestsMap:  map[string]*networkv1.NetworkAttachmentDefinition{},
+			expectedNetworksRequests: []string{""},
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/pod_test.go:28]: Please check if this value is legal in multus.

File: pkg/internal/checkup/pod/pod_test.go [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+func newNetAttachDef(namespace, name string) *networkv1.NetworkAttachmentDefinition {
+	return &networkv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/pod_test.go:61]: Could you please move this helper to the bottom of the file?

File: pkg/internal/client/client.go [hunk 15]
```diff
@@ -21,16 +21,22 @@ package client
 
 import (
 	"context"
+	"log"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8scorev1 "k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (orelmisan) [inline pkg/internal/client/client.go:27]: nit: These were previously defined as "corev1" and "metav1".

File: pkg/internal/client/client.go [hunk 16]
```diff
@@ -21,16 +21,22 @@ package client
...
 	"k8s.io/client-go/rest"
 
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
```
Comment (orelmisan) [inline pkg/internal/client/client.go:30]: nit: this was previously defined as "networkv1"

File: pkg/internal/client/client.go [hunk 17]
```diff
@@ -21,16 +21,22 @@ package client
...
+
 	kvcorev1 "kubevirt.io/api/core/v1"
+	netattdefclient "kubevirt.io/client-go/generated/network-attachment-definition-client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
```
Comment (orelmisan) [inline pkg/internal/client/client.go:33]: Please consider using the client from https://github.com/k8snetworkplumbingwg/network-attachment-definition-client.

File: pkg/internal/client/client.go [hunk 18]
```diff
@@ -49,7 +55,27 @@ func New() (*Client, error) {
 		return nil, err
 	}
 
-	return &Client{client}, nil
+	cniClient, err := netattdefclient.NewForConfig(config)
```
Comment (orelmisan) [inline pkg/internal/client/client.go:58]: Why was the name cniClient was chosen?

File: pkg/internal/client/client.go [hunk 19]
```diff
@@ -49,7 +55,27 @@ func New() (*Client, error) {
...
+}
+
+func (c *Client) CreatePod(ctx context.Context, namespace string, pod *k8scorev1.Pod) (*k8scorev1.Pod, error) {
+	log.Printf("Creating Pod %s/%s..", namespace, pod.Name)
+	return c.KubevirtClient.CoreV1().Pods(namespace).Create(ctx, pod, k8smetav1.CreateOptions{})
+}
+
+func (c *Client) GetPod(ctx context.Context, namespace, name string) (*k8scorev1.Pod, error) {
+	return c.KubevirtClient.CoreV1().Pods(namespace).Get(ctx, name, k8smetav1.GetOptions{})
 }
```
Comment (orelmisan) [inline pkg/internal/client/client.go:79]: Could you please add the pod functionality in a different PR?

File: pkg/internal/checkup/checkup.go [hunk 20]
```diff
@@ -47,6 +47,7 @@ type kubeVirtVMIClient interface {
 type Checkup struct {
 	client    kubeVirtVMIClient
 	namespace string
+	params    config.Config
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go]: Could you please add an explanation in the commit message regarding why is it necessary?
For example: Cannot do my operations in new, and need to perform them in setup.

Please change the subject's prefix from mainflow to checkup.

File: pkg/internal/checkup/checkup.go [hunk 21]
```diff
@@ -25,12 +25,16 @@ import (
 	"log"
 	"time"
 
+	k8scorev1 "k8s.io/api/core/v1"
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:28]: This was previously "corev1"

File: pkg/internal/client/client.go [hunk 22]
```diff
@@ -49,7 +55,27 @@ func New() (*Client, error) {
...
+}
+
+func (c *Client) GetNetworkAttachmentDefinition(
+	ctx context.Context,
+	namespace, name string) (*netattdefv1.NetworkAttachmentDefinition, error) {
+	return c.K8sCniCncfIoV1Interface.NetworkAttachmentDefinitions(namespace).Get(ctx, name, k8smetav1.GetOptions{})
+}
+
+func (c *Client) CreatePod(ctx context.Context, namespace string, pod *k8scorev1.Pod) (*k8scorev1.Pod, error) {
+	log.Printf("Creating Pod %s/%s..", namespace, pod.Name)
```
Comment (orelmisan) [inline pkg/internal/client/client.go:73]: Please consider taking out this log printing.

File: pkg/internal/checkup/checkup.go [hunk 23]
```diff
@@ -183,3 +201,86 @@ func randomizeName(prefix string) string {
...
+		var err error
+		updatedPod, err = c.client.GetPod(ctx, namespace, name)
+		if err != nil {
+			return false, err
+		}
+		return pod.PodInRunningPhase(updatedPod), nil
+	}
+	const interval = time.Second * 5
+	if err := wait.PollImmediateUntilWithContext(ctx, interval, conditionFn); err != nil {
+		return nil, fmt.Errorf("failed to wait for POD '%s/%s' to be in Running Phase: %v", namespace, name, err)
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:247]: Please consider using the ObjectFullName function.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
Please see the inline comments.

File: pkg/internal/checkup/pod/multus.go (hunk 8 above)
Comment (orelmisan) [reply pkg/internal/checkup/pod/multus.go:11]: On second thought, if the assosiation is already known to the caller, maybe its better to pass in the selectors from the outside.

File: tests/checkup_test.go (hunk 1 above)
Comment (RamLavi) [reply tests/checkup_test.go:226]: DONE

File: pkg/internal/checkup/pod/container.go (hunk 5 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/container.go:1]: done for entire package modules.

File: pkg/internal/checkup/pod/container.go (hunk 6 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/container.go:97]: It's basically the same, but this way it's more visible so I prefer that.

File: pkg/internal/checkup/pod/multus.go (hunk 7 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/multus.go:1]: DONE

File: pkg/internal/checkup/pod/multus.go (hunk 8 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/multus.go:11]: yep..

File: pkg/internal/checkup/pod/pod_test.go (hunk 9 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/pod_test.go:1]: DONE

File: pkg/internal/checkup/pod/pod_test.go (hunk 11 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/pod_test.go:7]: DONE

File: pkg/internal/checkup/pod/pod_test.go (hunk 10 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/pod_test.go:20]: will keep this as agreed above

File: pkg/internal/checkup/pod/pod_test.go (hunk 13 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/pod_test.go:28]: I tried it on a pod and it appears that both "" and "[]" are valid inputs to this annotation. But I think "[]" is more correct so I'll use that.

File: pkg/internal/checkup/pod/pod_test.go (hunk 14 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/pod_test.go:61]: DONE

File: pkg/internal/checkup/pod/spec.go (hunk 2 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/spec.go:1]: DONE

File: pkg/internal/checkup/pod/spec.go (hunk 3 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/spec.go:120]: DONE

File: pkg/internal/checkup/pod/spec.go (hunk 4 above)
Comment (RamLavi) [reply pkg/internal/checkup/pod/spec.go:125]: You're right. Removed. thanks!

File: pkg/internal/client/client.go (hunk 15 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:27]: yes I changed it to be be more unified. After this PR I'll recheck if I missed anything and unify the entire repo.

File: pkg/internal/client/client.go (hunk 16 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:30]: DONE

File: pkg/internal/client/client.go (hunk 17 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:33]: good idea. DONE

File: pkg/internal/client/client.go (hunk 18 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:58]: changed to nadClient

File: pkg/internal/client/client.go (hunk 22 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:73]: DONE

File: pkg/internal/client/client.go (hunk 15 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:27]: I'll return it and change it in another PR

File: pkg/internal/client/client.go (hunk 19 above)
Comment (RamLavi) [reply pkg/internal/client/client.go:79]: DONE

File: pkg/internal/checkup/checkup.go (hunk 20 above)
Comment (RamLavi) [reply pkg/internal/checkup/checkup.go]: DONE

File: pkg/internal/checkup/checkup.go (hunk 21 above)
Comment (RamLavi) [reply pkg/internal/checkup/checkup.go:28]: will unify it in a separate PR.

File: pkg/internal/checkup/checkup.go (hunk 23 above)
Comment (RamLavi) [reply pkg/internal/checkup/checkup.go:247]: DONE

File: pkg/internal/checkup/pod/container.go [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+func WithContainerHugepagesVolumeMount() ContainerOption {
+	return func(container *corev1.Container) {
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      "hugepages",
+			MountPath: "/mnt/huge",
+		})
+	}
+}
+
+func NewSecurityContext(user int64, allowPrivilegeEscalatio bool, addCapabilities []corev1.Capability) *corev1.SecurityContext {
```
Comment (orelmisan) [inline pkg/internal/checkup/pod/container.go:110]:
```suggest
func NewSecurityContext(user int64, allowPrivilegeEscalation bool, addCapabilities []corev1.Capability) *corev1.SecurityContext {
```

File: pkg/internal/checkup/pod/pod_test.go (hunk 11 above)
Comment (orelmisan) [reply pkg/internal/checkup/pod/pod_test.go:30]: Is is still not done :)
Please also consider to replace it to:
```go
assert "github.com/stretchr/testify/require"
```

File: pkg/internal/checkup/checkup.go [hunk 25]
```diff
@@ -183,3 +201,89 @@ func randomizeName(prefix string) string {
...
+	const interval = time.Second * 5
+	if err := wait.PollImmediateUntilWithContext(ctx, interval, conditionFn); err != nil {
+		return nil, fmt.Errorf("failed to wait for POD '%s' to be in Running Phase: %v", podFullName, err)
+	}
+
+	log.Printf("Pod %s is Running", podFullName)
+	return updatedPod, nil
+}
+
+func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest string) *k8scorev1.Pod {
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:257]: Is the full checkup config struct required here?

File: pkg/internal/checkup/checkup.go [hunk 26]
```diff
@@ -183,3 +201,89 @@ func randomizeName(prefix string) string {
...
 }
+
+func (c *Checkup) createAndRunTrafficGeneratorPod(ctx context.Context) error {
+	netAttachDef, err := c.client.GetNetworkAttachmentDefinition(
+		ctx,
+		c.namespace,
+		c.params.NetworkAttachmentDefinitionName)
+	if err != nil {
+		return err
+	}
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:212]: We can remove this call to get the NetworkAttachmentDefinition.
Its existence only matters to multus.

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1674983586

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [45.628 seconds]
------------------------------

Ran 1 of 1 Specs in 45.633 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (45.63s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	45.664s

```

File: pkg/internal/checkup/checkup.go [hunk 27]
```diff
@@ -63,6 +73,18 @@ func New(client kubeVirtVMIClient, namespace string, checkupConfig config.Config
...
 func (c *Checkup) Setup(ctx context.Context) error {
+	const errMessagePrefix = "setup"
+	var err error
+
+	c.trafficGeneratorPod, err = c.createTrafficGeneratorPod(ctx)
+	if err != nil {
+		return fmt.Errorf("%s: %w", errMessagePrefix, err)
+	}
+
+	if c.trafficGeneratorPod, err = c.waitForPodRunningStatus(ctx, c.namespace, c.trafficGeneratorPod.Name); err != nil {
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:84]: Please take the updated pod to a new variable in order to not assign nil to `c.trafficGeneratorPod` in case of an error.

This could cause a problem to remove the Pod on teardown, in case of a failure to wait for the Pod.

Comment (RamLavi) [reply pkg/internal/checkup/checkup.go]: DONE

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1674984300

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [35.569 seconds]
------------------------------

Ran 1 of 1 Specs in 35.573 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (35.57s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	35.605s
```

**Review by orelmisan — APPROVED** [review]
Thank you for the changes.
Lets consider in a follow-up PR breaking the function that creates the Pod.
